test/externalTests/gp2.sh

```diff
@@ -67,6 +67,9 @@ function gp2_test
     force_hardhat_unlimited_contract_size "$config_file" "$config_var"
     yarn
 
+    # Remove the config section that requires an Etherscan key. We don't need it just to run tests.
+    sed -i '/^  etherscan: {$/,/^  },$/d' hardhat.config.ts
```

Comment:
Perhaps this could be improved to not rely on fixed white spaces. But, I suppose it's not possible to reliably capture the closing curly brackets using regex,  given the possibility of ambiguity in the case of nested objects..

```suggestion
    sed -i '/^\s*etherscan:\s*{$/,/^  },$/d' hardhat.config.ts
```

Reply:
Yeah, the closing brace is exactly the reason why I went for hard-coded whitespace and I'd keep it that way for simplicitly. I just wanted to avoid having to actually parse it or doing some complex thing with multiline regexes (those are always a problem in sed).

So I'd keep it as is because if the indentation changes here, it probably means it has to be adjusted anyway.